org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/xsi/XSIFormatterExperimentalTest.java

```diff
@@ -101,53 +163,140 @@ public void xsiSchemaLocationSplitOnPair() throws BadLocationException {
+				"<beans " + //
+				"xmlns=\"http://www.springframework.org/schema/beans\" " + //
+				"xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" " + //
+				"xmlns:util=\"http://www.springframework.org/schema/util\" " + //
+				"xsi:schemaLocation=";
+		assertFormat(content, content, settings);
+
+		content = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans " + //
+				"xmlns=\"http://www.springframework.org/schema/beans\" " + //
+				"xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" " + //
+				"xmlns:util=\"http://www.springframework.org/schema/util\" " + //
+				"xsi:schemaLocation=\"";
+		assertFormat(content, content, settings);
+
+		content = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans " + //
+				"xmlns=\"http://www.springframework.org/schema/beans\" " + //
+				"xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" " + //
+				"xmlns:util=\"http://www.springframework.org/schema/util\" " + //
+				"xsi:schemaLocation=\"\"";
+		assertFormat(content, content, settings);
 	}
 
-	@Disabled
 	@Test
 	public void xsiSchemaLocationSplitOnElementWithTabs() throws BadLocationException {
 		SharedSettings settings = createSettings();
 		settings.getFormattingSettings().setTabSize(4);
 		settings.getFormattingSettings().setInsertSpaces(false);
 		XSISchemaLocationSplit.setSplit(XSISchemaLocationSplit.onElement, settings.getFormattingSettings());
+		String content = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans\r\n" + //
+				"    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xsi:schemaLocation=\"\r\n" + //
+				"        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd\r\n"
+				+ //
+				"        http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd\">\r\n"
+				+ //
+				"\r\n" + //
+				"</beans>";
+		String expected = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"<beans xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.springframework.org/schema/beans\r\n"
+				+ //
+				"																				 http://www.springframework.org/schema/beans/spring-beans.xsd\r\n"
+				+ //
+				"																				 http://www.springframework.org/schema/util\r\n"
+				+ //
+				"																				 http://www.springframework.org/schema/util/spring-util.xsd\"> </beans>";
+		assertFormat(content, expected, settings,
+				te(1, 6, 2, 4, " "), //
+				te(2, 57, 3, 4, " "), //
+				te(3, 24, 4, 8, ""), //
+				te(4, 51, 4, 52, "\r\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t "), //
+				te(4, 112, 5, 8, "\r\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t "), //
+				te(5, 50, 5, 51, "\r\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t "), //
+				te(5, 111, 7, 0, " "));
+		assertFormat(expected, expected, settings);
+	}
+
+	@Test
+	public void xsiSchemaLocationSplitOnElementWithTabsWithSplitAttribute() throws BadLocationException {

```

Comment:
It should be good to have a test too with setInsertSpaces to false. You could copy paste this test and set setInsertSpacesto false.